crates/relayer/src/keyring/errors.rs

```diff
@@ -112,5 +111,29 @@ define_error! {
             |e| {
                 format!("invalid HD path: {0}", e.path)
             },
+
+        AddressTypeNotFound
+            {
+                address: Vec<u8>,
+                public_key: Vec<u8>,
+            }
+            |_| {
+                "No address type found that matches the public key"
```

Comment:
Let's show the address and the public key in the error message? Otherwise the fields won't be used at all.